packages/core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -119,6 +122,23 @@ export const DatePicker: React.FC<Props> & WithStyle = React.memo(
             </DateIcon>
         );
 
+        const sharedProps = {
+            errorText: errorText || builtInErrorMessage,
+            id: id,
+            ref: inputRef,
+            required: required,
+            fullWidth,
+            size: size,
+            disabled: disabled,
+            value: textValue,
+            onChange: onTextChange,
+            ...{ ...restProps, onBlur, onFocus, onInvalid }
+        };
+        const mobileProps = { type: 'date' };
+        const desktopProps = {
+            suffix: suffixEl,
+            mask: displayFormat.replace(new RegExp('\\/|\\-', 'g'), ' $& ').toUpperCase()
+        };
```

Comment:
This looks good to me - I think this is a clear and more concise way to accomplish things.